### PR TITLE
LPS-75855

### PR DESCRIPTION
**Pull request by jonmak08**
Hey @shuyangzhou,

Attached is an update for http://issues.liferay.com/browse/LPS-75855.  Would you mind taking a look at this?  I know it is not related to you, but it relates to backend and the filters which I think you might have better insight into than me.

Please let me know if you have any questions. Thanks!

### CODE REVIEW

File: portal-impl/src/com/liferay/portal/servlet/filters/aggregate/ServletPaths.java [hunk 1]
```diff
@@ -71,8 +74,26 @@ public ServletPaths down(String path) {
 			return this;
 		}
 
-		return new ServletPaths(
-			_resourcePath.concat(normalizedPath), _servletContext);
+		int pos = normalizedPath.indexOf(CharPool.QUESTION);
+
+		if (pos != -1) {
+			normalizedPath = normalizedPath.substring(0, pos);
```
Comment (shuyangzhou) [inline portal-impl/src/com/liferay/portal/servlet/filters/aggregate/ServletPaths.java]: I don't think this is right to simply throw away the query string.
For normalized path building, yes, we need to remove the query string. But shouldn't we add it back to the final normalized path?

Comment (jonmak08) [reply portal-impl/src/com/liferay/portal/servlet/filters/aggregate/ServletPaths.java]: /cc @jonathanmccann @gregory-bretall, thoughts on @shuyangzhou's comment?

Comment (holatuwol) [reply portal-impl/src/com/liferay/portal/servlet/filters/aggregate/ServletPaths.java]: @shuyangzhou: We tried leaving it there, and Windows complained about ? being an invalid character inside of a filename.

Comment (holatuwol) [reply portal-impl/src/com/liferay/portal/servlet/filters/aggregate/ServletPaths.java]: @shuyangzhou: Or do you mean, add the query string to the return value `return new ServletPaths(downPathString, _servletContext)`?

Comment (gregory-bretall) [reply portal-impl/src/com/liferay/portal/servlet/filters/aggregate/ServletPaths.java]: @shuyangzhou and @holatuwol I have made a version of the fix that adds the query string back in to the return value and it still works in my testing, does that seem like something we want?

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/53331

Ok, thanks @holatuwol @gregory-bretall , I am taking in the query string removal version.

**Pull request closed without merging**